src/Adapter/Module/Repository/ModuleRepository.php

```diff
@@ -107,4 +112,32 @@ public function getActiveModulesPaths(): array
 
         return $this->activeModulesPaths;
     }
+
+    /**
+     * Returns an array of native modules
+     *
+     * @return array<string>
+     */
+    public function getNativeModules(): array
+    {
+        $content = file_get_contents(_PS_ROOT_DIR_ . '/composer.lock');
```

Comment:
This file might not exists

Reply:
It's the only file kept in production package.

PM have decided to use composer for checking if they are native.

Link : https://github.com/PrestaShop/PrestaShop/issues/9965#issuecomment-953870811